### PR TITLE
Adjustments for deploying extension on Garden runtime cluster by gardener-operator

### PR DESCRIPTION
**Pull request by oliver-goetz**
**How to categorize this PR?**

/area control-plane
/kind enhancement
/platform gcp

**What this PR does / why we need it**:
This PR enables the extension to be deployed via `extensions.operator.gardener.cloud` CRD. It includes minor changes in code and a refactoring of the Helm charts.

**Which issue(s) this PR fixes**:
Part of https://github.com/gardener/gardener/issues/9635

**Special notes for your reviewer**:

**Release note**:

```feature operator
The extension can now be deployed via `extensions.operator.gardener.cloud` CRD.
```
```breaking operator
`gardener-extension-admission-gcp` Helm chart has been removed. The admission can be deployed by applying `admission-gcp-application` and `admission-gcp-runtime` charts separately. With this change the `global` structure in Helm values of these charts has been removed. Still supported settings have been moved to other sections.
```

### CODE REVIEW

File: charts/gardener-extension-admission-gcp/charts/application/templates/rbac.yaml [hunk 1]
```diff
@@ -21,6 +21,34 @@ rules:
...
+  - watch
+  - update
+- apiGroups:
+  - events.k8s.io
+  resources:
+  - events
+  verbs:
+  - create
+  - patch
+  - update
```
Comment (MartinWeindel) [inline charts/gardener-extension-admission-gcp/charts/application/templates/rbac.yaml:51]: Looks as if these changes have been applied by mistake in the provider-aws.
Please also remove them. See https://github.com/gardener/gardener-extension-provider-aws/pull/1100#pullrequestreview-2447758364
```suggestion
```

File: charts/gardener-extension-provider-gcp/templates/deployment.yaml [hunk 2]
```diff
@@ -64,8 +68,14 @@ spec:
         - --webhook-config-namespace={{ .Release.Namespace }}
         - --webhook-config-service-port={{ .Values.webhookConfig.servicePort }}
         - --webhook-config-server-port={{ .Values.webhookConfig.serverPort }}
+        {{- if .Values.gardener.runtimeCluster.enabled }}
+        - --disable-controllers=backupentry,bastion,controlplane,infrastructure,worker,healthcheck
```
Comment (MartinWeindel) [inline charts/gardener-extension-provider-gcp/templates/deployment.yaml:72]: Improvement suggested by @timuthy 
```suggestion
        - --controllers=backupbucket,dnsrecord
```

File: cmd/gardener-extension-provider-gcp/app/app.go [hunk 3]
```diff
@@ -230,6 +230,7 @@ func NewControllerManagerCommand(ctx context.Context) *cobra.Command {
 			reconcileOpts.Completed().Apply(&gcpcontrolplane.DefaultAddOptions.IgnoreOperationAnnotation, &gcpcontrolplane.DefaultAddOptions.ExtensionClass)
 			reconcileOpts.Completed().Apply(&gcpworker.DefaultAddOptions.IgnoreOperationAnnotation, &gcpworker.DefaultAddOptions.ExtensionClass)
 			reconcileOpts.Completed().Apply(&gcpbastion.DefaultAddOptions.IgnoreOperationAnnotation, &gcpbastion.DefaultAddOptions.ExtensionClass)
+			reconcileOpts.Completed().Apply(&gcpdnsrecord.DefaultAddOptions.IgnoreOperationAnnotation, &gcpdnsrecord.DefaultAddOptions.ExtensionClass)
```
Comment (MartinWeindel) [inline cmd/gardener-extension-provider-gcp/app/app.go:233]: Extension class is also needed for `BackupBucket`.
```suggestion
			reconcileOpts.Completed().Apply(&gcpdnsrecord.DefaultAddOptions.IgnoreOperationAnnotation, &gcpdnsrecord.DefaultAddOptions.ExtensionClass)
			reconcileOpts.Completed().Apply(&gcpbackupbucket.DefaultAddOptions.IgnoreOperationAnnotation, &gcpbackupbucket.DefaultAddOptions.ExtensionClass)
```

**Review by MartinWeindel — CHANGES_REQUESTED** [review]
Thanks for the adjustments. 🙂
There are some latest changes in the similar PR for `provider-aws` applying also here.

**Review by AndreasBurger — APPROVED** [review]

**Pull request merged**